prism/src/main/java/org/apache/falcon/resource/proxy/ExtensionManagerProxy.java

```diff
@@ -380,51 +400,83 @@ public APIResult submitAndSchedule(
+                final HttpServletRequest bufferedRequest = getBufferedRequest(httpServletRequest);
+                final Set<String> colos = getApplicableColos(entity.getEntityType().toString(), entity);
+
+                new EntityProxy(entity.getEntityType().toString(), entity.getName()) {
+                    @Override
+                    protected Set<String> getColosToApply() {
+                        return colos;
+                    }
+
+                    @Override
+                    protected APIResult doExecute(String colo) throws FalconException {
+                        return getEntityManager(colo).invoke("schedule", bufferedRequest, entity.getEntityType().toString(),
+                            entity.getName(), colo, Boolean.FALSE, "");
+                    }
+                }.execute();
             }
         }
-        return feeds;
     }
 
-    private List<Entity> getProcesses(List<FormDataBodyPart> processForms) {
-        List<Entity> processes = new ArrayList<>();
-        if (processForms != null && !processForms.isEmpty()) {
-            for (FormDataBodyPart formDataBodyPart : processForms) {
-                processes.add(formDataBodyPart.getValueAs(Process.class));
-            }
+    private BufferedRequest getBufferedRequest(HttpServletRequest request) {
+        if (request instanceof BufferedRequest) {
+            return (BufferedRequest) request;
         }
-        return processes;
+        return new BufferedRequest(request);
     }
 
-    protected void submitEntities(String extensionName, String doAsUser, String jobName,
-                                  Map<EntityType, List<Entity>> entityMap, InputStream configStream)
-        throws FalconException, IOException {
+    protected void submitEntities(String extensionName, String jobName,
+                                  Map<EntityType, List<Entity>> entityMap, InputStream configStream,
+                                  HttpServletRequest request) throws FalconException, IOException, JAXBException {
         List<Entity> feeds = entityMap.get(EntityType.FEED);
         List<Entity> processes = entityMap.get(EntityType.PROCESS);
         validateFeeds(feeds);
         validateProcesses(processes);
         List<String> feedNames = new ArrayList<>();
         List<String> processNames = new ArrayList<>();
-        for (Entity feed : feeds) {
-            submitInternal(feed, doAsUser);
-            feedNames.add(feed.getName());
-        }
-        for (Entity process: processes) {
-            submitInternal(process, doAsUser);
-            processNames.add(process.getName());
+
+        for(Map.Entry<EntityType, List<Entity>> entry : entityMap.entrySet()){
+            for(final Entity entity : entry.getValue()){
+                final HttpServletRequest httpServletRequest = getEntityStream(entity, entity.getEntityType(), request);
```

Comment:
Since the eventual goal is to get the bufferedRequest I think getEntityStream and getBufferedRequest into a single method.